### PR TITLE
Rt1064 usb enabling

### PR DESCRIPTION
**Pull request by loicpoulain**

### CODE REVIEW

**Comment by MaureenHelm** [comment]
@MarkWangChinese please review

**Review by MaureenHelm — APPROVED** [review]

**Review by dbkinder — APPROVED** [review]
+1 for doc changes

**Review by finikorg — APPROVED** [review]

File: subsys/usb/class/netusb/function_eem.c [hunk 1]
```diff
@@ -221,11 +221,15 @@ static struct netusb_function eem_function = {
 	.send_pkt = eem_send,
 };
 
-static inline void eem_status_interface(const u8_t *iface)
+static inline void eem_status_interface(const u8_t *desc)
 {
+	const struct usb_if_descriptor *if_desc = (void *)desc;
+	u8_t iface_num = if_desc->bInterfaceNumber;
+	u8_t alt_set = if_desc->bAlternateSetting;
```
Comment (finikorg) [inline subsys/usb/class/netusb/function_eem.c]: You do not need `alt_set` here

File: drivers/usb/device/usb_dc_mcux_ehci.c [hunk 2]
```diff
@@ -33,11 +37,7 @@ extern void USB_DeviceEhciIsrFunction(void *deviceHandle);
 /* The minimum value is 1 */
 #define EP_BUF_NUMOF_BLOCKS     ((NUM_OF_EP_MAX + 3) / 4)
 /* The max MPS is 1023 for FS, 1024 for HS. */
-#if defined(CONFIG_NOCACHE_MEMORY)
-__nocache K_MEM_POOL_DEFINE(ep_buf_pool, 16, 1024, EP_BUF_NUMOF_BLOCKS, 4);
-#else
-K_MEM_POOL_DEFINE(ep_buf_pool, 16, 1024, EP_BUF_NUMOF_BLOCKS, 4);
-#endif
+K_MEM_POOL_DEFINE(ep_buf_pool, 16, 2048, EP_BUF_NUMOF_BLOCKS, 4);
```
Comment (MarkWangChinese) [inline drivers/usb/device/usb_dc_mcux_ehci.c]: Hi @loicpoulain There are SOC issues in the RT1050 or RT1060 that we cannot put the USB transfer buffer in cachable ram and use cache maintenance API (for example: DCACHE_InvalidateByRange) to operate USB transfer buffer. So I define the transfer buffer in the noncachable ram.  In the simple case, the issue may not be triggered, the issue may be triggered by mass bulk transfers etc. The issue is that the buffer may not updated rightly.

File: boards/arm/mimxrt1064_evk/mimxrt1064_evk.dts [hunk 3]
```diff
@@ -88,3 +88,8 @@ arduino_i2c: &i2c1 {};
 		status = "ok";
 	};
 };
+
+&usbd1 {
+	status = "ok";
+};
+
```
[the remark this answers is not in the document]
Comment (loicpoulain) [reply boards/arm/mimxrt1064_evk/mimxrt1064_evk.dts]: fixed

File: subsys/usb/class/netusb/function_eem.c (hunk 1 above)
Comment (loicpoulain) [reply subsys/usb/class/netusb/function_eem.c]: thanks fixed

File: drivers/usb/device/usb_dc_mcux_ehci.c (hunk 2 above)
Comment (loicpoulain) [reply drivers/usb/device/usb_dc_mcux_ehci.c]: It is a hardware issue which has been fixed? I do not reproduce it with RT1064 despite running in high-speed mode and stressing USB by sending DATA via ethernet over USB class. I changed the commit to take into account this issue.

Comment (MarkWangChinese) [reply drivers/usb/device/usb_dc_mcux_ehci.c]: It is not fixed yet and it is hard to re-produce.

Comment (loicpoulain) [reply drivers/usb/device/usb_dc_mcux_ehci.c]: @MarkWangChinese , coud you please reveiw again.

Comment (MarkWangChinese) [reply drivers/usb/device/usb_dc_mcux_ehci.c]: It is OK, thanks

**Pull request merged**